crates/matrix-sdk-ui/tests/integration/room_list_service.rs

```diff
@@ -3016,3 +3016,104 @@ async fn test_sync_indicator() -> Result<(), Error> {
 
     Ok(())
 }
+
+#[cfg(feature = "experimental-room-list-with-unified-invites")]
+#[async_test]
+async fn test_unified_invites_enabled() -> Result<(), Error> {
+    let (client, server) = logged_in_client_with_server().await;
+    let room_list = RoomListService::new_with_unified_invites(client, true).await?;
+
+    let sync = room_list.sync();
+    pin_mut!(sync);
+
+    sync_then_assert_request_and_fake_response! {
+        [server, room_list, sync]
+        states = Init => SettingUp,
+        assert request = {
+            "conn_id": "room-list",
+            "lists": {
+                ALL_ROOMS: {
+                    "ranges": [[0, 19]],
+                    "required_state": [
+                        ["m.room.avatar", ""],
+                        ["m.room.encryption", ""],
+                        ["m.room.member", "$LAZY"],
+                        ["m.room.member", "$ME"],
+                        ["m.room.power_levels", ""],
+                    ],
+                    "filters": {
+                        // The `is_invite` field should be missing from the ALL_ROOMS filters
+                        "is_tombstoned": false,
+                        "not_room_types": ["m.space"],
+                    },
+                    "bump_event_types": [
+                        "m.room.message",
+                        "m.room.encrypted",
+                        "m.sticker",
+                    ],
+                    "sort": ["by_recency", "by_name"],
+                    "timeline_limit": 1,
+                },
+                INVITES: {
```

Comment:
Strange that we have the invites list set up in unified all_rooms+invites mode, but that's preexisting, and something we can solve later on.